build_tools/cmake/iree_run_module_test.cmake

```diff
@@ -280,54 +276,15 @@ function(iree_benchmark_suite_module_test)
     return()
   endif()
 
-  if(DEFINED _RULE_MODEL)
-    string(TOUPPER "${_PLATFORM}" _UPPER_PLATFORM)
-    set(_IREE_MODULE_COMPILE_CONFIG_ID "${IREE_MODULE_COMPILE_CONFIG_ID_${_UPPER_PLATFORM}}")
-    if("${_IREE_MODULE_COMPILE_CONFIG_ID}" STREQUAL "")
-      message(WARNING "No compile config for ${_PLATFORM}. Skip ${_RULE_MODEL}.")
-      return()
-    endif()
-    # Drop the UUID prefix ".{8}-.{4}-.{4}-.{4}-.{12}_", 37 characters in total.
-    string(SUBSTRING "${_RULE_MODEL}" 37 -1 _MODEL_NAME)
-    set(_SRC "${IREE_BENCHMARK_SUITE_DIR}/iree/${_RULE_MODEL}/${_IREE_MODULE_COMPILE_CONFIG_ID}/${_MODEL_NAME}.vmfb")
-  else()
-    set(_MODULE_FLAG_DIR "${IREE_BENCHMARK_SUITE_DIR}/${_RULE_BENCHMARK_MODULE_SRC}/")
-    # Find the platform specific module flag file with matching path name.
-    # TODO(#10391): Update this logic with the new benchmark framework.
-    if(_PLATFORM STREQUAL "riscv64-Linux")
-      set(_FLAGFILE_HINT_PATH "${_MODULE_FLAG_DIR}/iree-llvm-cpu*RV64*__full-inference,default-flags/flagfile")
-    elseif(_PLATFORM STREQUAL "riscv32-Linux")
-      set(_FLAGFILE_HINT_PATH "${_MODULE_FLAG_DIR}/iree-llvm-cpu*RV32*__full-inference,default-flags/flagfile")
-    elseif(_PLATFORM STREQUAL "android-arm64-v8a")
-      set(_FLAGFILE_HINT_PATH "${_MODULE_FLAG_DIR}/iree-llvm-cpu*ARM64-v8A*__big-core,full-inference,default-flags/flagfile")
-    else()  # X86_64
-      set(_FLAGFILE_HINT_PATH "${_MODULE_FLAG_DIR}/iree-llvm-cpu*x86_64*__full-inference,default-flags/flagfile")
-    endif()
-    file(GLOB _FLAGFILE_PATH
-        LIST_DIRECTORIES FALSE
-        "${_FLAGFILE_HINT_PATH}"
-      )
-    if(NOT _FLAGFILE_PATH)
-      message(SEND_ERROR "Could not locate flagfile matching '${_FLAGFILE_HINT_PATH}' for ${_RULE_BENCHMARK_MODULE_SRC}")
-      return()
-    endif()
-
-    list(LENGTH _FLAGFILE_PATH _FLAGFILE_COUNT)
-    if(_FLAGFILE_COUNT GREATER 1)
-      message(SEND_ERROR "Found multiple files matching '${_FLAGFILE_HINT_PATH}' for ${_RULE_BENCHMARK_MODULE_SRC}: ${_FLAGFILE_PATH}")
-    endif()
-
-    cmake_path(GET _FLAGFILE_PATH PARENT_PATH _FLAG_FILE_DIR)
-    file(STRINGS "${_FLAGFILE_PATH}" _FLAGS ENCODING UTF-8)
-    # Parse the flagfile to find the vmfb location.
-    # TODO(#10391): Update this logic with the new benchmark framework.
-    foreach(_FLAG ${_FLAGS})
-      if(_FLAG MATCHES "--module_file=")
-        string(REPLACE "--module_file=" "" _SRC "${_FLAG}")
-        set(_SRC "${_FLAG_FILE_DIR}/${_SRC}")
-      endif()
-    endforeach(_FLAG)
+  string(TOUPPER "${_PLATFORM}" _UPPER_PLATFORM)
+  set(_IREE_MODULE_COMPILE_CONFIG_ID "${IREE_MODULE_COMPILE_CONFIG_ID_${_UPPER_PLATFORM}}")
+  if("${_IREE_MODULE_COMPILE_CONFIG_ID}" STREQUAL "")
+    message(WARNING "No compile config for ${_PLATFORM}. Skip ${_RULE_MODEL}.")
```

Comment:
I don't like warnings (there's a place for them, but should generally be avoided IMO). Should this be an error? What's the use case for creating this rule and then not having a compilation configuration?

Reply:
This is due to the current non-ideal way to integrate e2e model tests and the test framework (see #11136). Essentially right now I try to change a little to bridge them together. I'm planning to refactor this later.

Reply:
Ok, I guess. Consider using INFO if this is something that is likely to show up frequently.

Reply:
Actually, if the target sets the `UNSUPPORTED_PLATFORMS` correctly, we should not hit this message at all.

Change this to `SEND_ERROR` could ensure the target is set appropriately, actually.

Reply:
That sounds a lot better to me...

Reply:
Changed to SEND_ERROR